### DISCUSSION TITLE
keyboard won't work

### DISCUSSION THREAD

**Discussion by nbonfire** [antonym/netboot.xyz#868] [Q&A] [answered]
I have a new machine I'm trying to image using netboot. I've used it successfully on other machines, but this one the keyboard won't respond once in netboot.xyz. The keyboard does work in the UEFI/boot menu, and net booting gets into the netboot.xyz menu (I see the countdown next to local disk), but after that no keys work (caps lock/num lock lights don't even change - num lock is on by default and its light stays on,)

Any ideas? I've tried multiple (known good) keyboards and plugged into different USB ports. The logic board has a PS/2 port, but I haven't had a PS/2 keyboard in over a decade 😅 

If it matters, I'm using the linuxserver docker container, with the 2.0.35 release

**Comment by antonym** [comment]
I would suggest trying the stock version of iPXE to see if you can get the keyboard working there first.  I have seen where the stack doesn't play well with USB sometimes and it manifests as the keyboard not working.

**Comment by Corruptsector** [comment]
Im having the exact same issue. Did you ever manage to figure it out?

**Reply by nbonfire** [reply]
sorry, I don't remember what finally fixed it and the server I was running it on is decommissioned now. I probably did what antonym suggested though.

**Comment by Firefishy** [comment]
Enabling "Enable persistent mouse and keyboard" under iLO: Power Management -> Power Settings unfortunately didn't work for me. 

Fresh boot with "ipmitool sol" console did work.

**Reply by NiKiZe** [reply]
You might be hitting https://github.com/netbootxyz/netboot.xyz/issues/1769 Where an USB keyboard is not detected by the EFI firmware drivers after iPXE takes over the USB bus, known to be an issue in AMI and Phoenix based firmware.

**Comment by antonym** [comment, marked as answer]
https://netboot.xyz/docs/kb/hardware/usb-keyboard/ Please view this and try.
